sui_core/src/authority/authority_store.rs

```diff
@@ -1074,3 +1104,28 @@ impl<const A: bool, const B: bool, S: Eq + Serialize + for<'de> Deserialize<'de>
             }))
     }
 }
+
+// The primary key type for object storage.
+#[serde_as]
+#[derive(Eq, PartialEq, Clone, Copy, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+struct ObjectKey(pub ObjectID, pub VersionNumber);
```

Comment:
I originally had it in messages.rs but its really an implementation detail of the store for now, so I moved it back in here. I think having it public would be confusing since it won't be clear to people when they should use ObjectKey vs ObjectRef